specification/sql/resource-manager/Microsoft.Sql/preview/2020-02-02-preview/HybridLinks.json

```diff
@@ -0,0 +1,460 @@
+          }
+        }
+      }
+    }
+  },
+  "definitions": {
+    "HybridLinkListResult": {
+      "description": "A list of hybrid links in instance.",
+      "type": "object",
+      "properties": {
+        "value": {
+          "description": "Array of results.",
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/HybridLink"
+          },
+          "readOnly": true
+        },
+        "nextLink": {
+          "description": "Link to retrieve next page of results.",
+          "type": "string",
+          "readOnly": true
+        }
+      }
+    },
+    "HybridLinkProperties": {
+      "description": "The properties of a hybrid link.",
+      "type": "object",
+      "properties": {
+        "targetDatabase": {
+          "description": "The name of the target database",
+          "type": "string"
+        },
+        "sourceEndpoint": {
+          "description": "The source endpoint",
+          "type": "string"
+        },
+        "primaryAvailabilityGroupName": {
+          "description": "The primary availability group name",
+          "type": "string"
+        },
+        "secondaryAvailabilityGroupName": {
+          "description": "The secondary availability group name",
+          "type": "string"
+        },
+        "replicationMode": {
+          "description": "The replication mode of hybrid link. Parameter will be ignored during link creation.",
+          "enum": [
+            "Async",
+            "Sync"
+          ],
+          "type": "string",
+          "x-ms-enum": {
+            "name": "ReplicationMode",
+            "modelAsString": true
+          }
+        },
+        "distributedAvailabilityGroupId": {
+          "format": "uuid",
+          "description": "The distributed availability group id",
+          "type": "string",
+          "readOnly": true
+        },
+        "sourceReplicaId": {
+          "format": "uuid",
+          "description": "The source replica id",
+          "type": "string",
+          "readOnly": true
+        },
+        "targetReplicaId": {
+          "format": "uuid",
+          "description": "The target replica id",
+          "type": "string",
+          "readOnly": true
+        }
+      }
+    },
+    "Resource": {
+      "description": "ARM resource.",
+      "type": "object",
+      "properties": {
+        "id": {
+          "description": "Resource ID.",
+          "type": "string",
+          "readOnly": true
+        },
+        "name": {
+          "description": "Resource name.",
+          "type": "string",
+          "readOnly": true
+        },
+        "type": {
+          "description": "Resource type.",
+          "type": "string",
+          "readOnly": true
+        }
+      },
+      "x-ms-azure-resource": true
+    },
+    "ProxyResource": {
+      "description": "ARM proxy resource.",
+      "type": "object",
+      "allOf": [
+        {
+          "$ref": "#/definitions/Resource"
```

Comment:
>"$ref": "#/definitions/Resource" [](start = 10, length = 32)

You are removing the required field from the resource and that's why pointing to common field will help making it correct,